src/BottomNavigation/BottomNavigationItem.js

```diff
@@ -34,54 +40,50 @@ function getStyles(props, context) {
   return styles;
 }
 
-class BottomNavigationItem extends React.Component {
-  static propTypes = {
-    /**
-     * Set the icon representing the view for this item.
-     */
-    icon: React.PropTypes.node,
-    /**
-     * Set the label describing the view for this item.
-     */
-    label: React.PropTypes.node,
-    /**
-     * Override the inline-styles of the root element.
-     */
-    style: React.PropTypes.object,
-  };
+const BottomNavigationItem = (props, context) => {
+  const {
+    label,
+    icon,
+    style,
+    ...other,
+  } = props;
 
-  static contextTypes = {
-    muiTheme: React.PropTypes.object.isRequired,
-  };
+  const {prepareStyles} = context.muiTheme;
+  const styles = getStyles(props, context);
 
-  render() {
-    const {
-      label,
-      icon,
-      style,
-      ...other,
-    } = this.props;
+  const styledIcon = cloneElement(icon, {
+    style: Object.assign({}, styles.icon, icon.props.style),
+    color: icon.props.color || styles.iconColor,
```

Comment:
Is it generally your policy to treat all falsy values as "no value"?

For `undefined` and `null` and `""`, it sort of makes sense to ignore them; but then again, I might want the component to complain that I am passing it junk, in case that's an indication of a bug on my end.

When I consider that a "bad" value of `1` will pass this test and fail prop validation, but a "bad" value of `0` will fail this test use the default icon color, I come to the usual conclusion that falsiness should be avoided. Hence why I used `Object.prototype.hasOwnProperty.call(icon.props, 'color')`, which is probably the truest way of detecting whether "the user passed a 'color' prop." Even if it is long and ugly.